src/cpp/KnowledgeItem.h

```diff
@@ -29,6 +30,7 @@ class KnowledgeItem {
 
 	virtual void save_to(BaseStorageSaver& storage);
 
+    virtual double match(SearchCriteria<std::string, std::time_t> *criteria);

```

Comment:
KnowledgeItem is expected to know all possible types supported by all its subclasses?
Now it is just string and Timestamp (std::time_t), but what if we need something else in future?